src/main/room/main/G1HRPS/RoomManager.java

```diff
@@ -1,97 +1,305 @@
+ */
+public void CheckOutAllGuest(String roomNumber) {
+	for(var i=0; i<room_list_.size();i++) {
+		if (room_list_.get(i).getRoomNumber()==roomNumber) {
+			room_list_.get(i).clearGuests();
+			room_list_.get(i).setStatus(Vacant);		
+			break;
+		}
+		else
+			System.out.println("room number does not exist");
 	}
-
-	/**
-	 * 
-	 * @param room
-	 * @param status
-	 */
-	public void SetRoomStatus(Room room, RoomStatus status) {
-		// TODO - implement RoomManager.SetRoomStatus
-		throw new UnsupportedOperationException();
+	throw new UnsupportedOperationException();
+}
+/**
+ * Removes one guest element from the guest list
+ * @param roomNumber This is the room number to remove guest element from
+ * @param guestName This is the guest name of guest element to be removed
+ */
+public void CheckOutOneGuest(String roomNumber,String guestName) {//note: guest Name required from user is full name of guest
+	for(var i=0; i<room_list_.size();i++) { 
+		if (room_list_.get(i).getRoomNumber()==roomNumber) {   //if room matches room number
+			for(var j=0; j<room_list_.get(i).getGuestList().size();j++) {
+				if(room_list_.get(i).getGuestList().get(j).getName==guestName) { // if guest name matches guest
+					room_list_.get(i).checkOutGuest(guestName);
+					break;
+				}
+				if (room_list_.get(i).getGuestList().size()==0) { // if guest list is empty, set room status to vacant
+					room_list_.get(i).setStatus(Vacant);
+				}
+			}
+		
+		}
 	}
+}
+/**
+ * Sets a room status to vacant, occupied, reserved, maintenance 
+ * This method is primarily to set room status to reserved and maintenance,
+ * as rooms are auto set to vacant and occupied in check in and check out.
+ * @param roomNumber This is room number of the room that status will be changed
+ * @param status This is the status to be changed into
+ */
+public void SetRoomStatus(String roomNumber, RoomStatus status) {
+	for(var i=0; i<room_list_.size();i++) {
+		if (room_list_.get(i).getRoomNumber()==roomNumber) {
+			room_list_.get(i).setStatus(status);
+		}
+	throw new UnsupportedOperationException();
+}
+}
+
+/**
+ * All room numbers in room_list_, organized by room type .
+ * e.g. Standard: Number: 2 out of 15 occupied 
+ * 				  Rooms: 02-21,02-22,02-33,02-44,02-55
+ * 		VIP:      Number: 3 out of 5 occupied
+ */
+public void GetRoomStatisticsByTypeOccupancyRate() {
+	int singleTotalCount;
+	int singleVacantCount;
+	String singleRoomNumber="";
+	int standardTotalCount;
+	int standardVacantCount;
+	String standardRoomNumber="";
+	int deluxeTotalCount;
+	int deluxeVacantCount;
+	String deluxeRoomNumber="";
+	int suiteTotalCount;
+	int suiteVacantCount;
+	String suiteRoomNumber="";
+	int VIPTotalCount;
+	int VIPVacantCount;
+	String VIPRoomNumber="";
+	
+	for(var i=0; i<room_list_.size();i++) {
+		RoomType type = room_list_.get(i).getRoomType();
+		switch(type) { //This switch block acts as a counter for different room types
+		
+		case Single:
+			singleTotalCount+=1;
```

Comment:
if you're using vs code, i suggest using Format Document to help you space out your code and make it more readable. otherwise it's fine, i can do it after combining.

Reply:
Okay, done so with eclipse IDE formatting